spyder/plugins/completion/languageserver/client.py

```diff
@@ -322,9 +323,22 @@ def send(self, method, params, kind):
             }
 
         logger.debug('{} request: {}'.format(self.language, method))
-        self.zmq_out_socket.send_pyobj(msg)
-        self.request_seq += 1
-        return int(_id)
+        # Try sending a message. If the send queue is full, keep trying for a
+        # a second before giving up.
+        timeout = 1
+        start_time = time.time()
+        timeout_time = start_time + timeout
+        while True:
+            try:
+                self.zmq_out_socket.send_pyobj(msg, flags=zmq.NOBLOCK)
+                self.request_seq += 1
+                return int(_id)
+            except zmq.error.Again:
+                if time.time() > timeout_time:
+                    raise
```

Comment:
Please don't `raise` here. Instead, change this for:

```python
self.sig_lsp_down.emit(self.language)
break
```

That will show a proper error message in a QMessageBox.